pkg/limits/frontend/frontend.go

```diff
@@ -34,25 +41,38 @@ type Frontend struct {
 	subservicesWatcher *services.FailureWatcher
 
 	limits IngestLimitsService
+
+	lifecycler        *ring.Lifecycler
```

Comment:
The frontend starting, stopping and running refs to lifecycler are missing:
- starting: See [IngestLimits.starting](https://github.com/grafana/loki/compare/main...feat/usage-tracker#diff-ca846946ac537045b180cd9bb04424c4d454ac70b15980f43d2fa275ca9feecaR122-R143)
- stopping: See [IngestLimits.stopping](https://github.com/grafana/loki/compare/main...feat/usage-tracker#diff-ca846946ac537045b180cd9bb04424c4d454ac70b15980f43d2fa275ca9feecaR238-R251)
- running: See [IngestLimits.running](https://github.com/grafana/loki/compare/main...feat/usage-tracker#diff-ca846946ac537045b180cd9bb04424c4d454ac70b15980f43d2fa275ca9feecaR157-R158)

Reply:
This seems to work. I've just been testing it in dev. I was able to observe new pods being registered as Pending, then Active when scaling up, and seeing them go down and come back following a rollout.